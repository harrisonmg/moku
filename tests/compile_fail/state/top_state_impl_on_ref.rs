use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    #[derive(Debug, Clone, Copy)]
    enum BlinkyState {}
    impl StateEnum for BlinkyState {}

    struct Top {}
    impl TopState<BlinkyState> for &Top {}
}

fn main() {}
