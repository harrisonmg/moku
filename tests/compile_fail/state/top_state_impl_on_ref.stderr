error: `moku::TopState` must be implemented on a plain struct
       You may also use a type alias: `type MyTopState = Option<bool>;`
  --> tests/compile_fail/state/top_state_impl_on_ref.rs:11:36
   |
11 |     impl TopState<BlinkyState> for &Top {}
   |                                    ^
